pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/SegmentPreprocessThrottler.java

```diff
@@ -18,208 +18,40 @@
-          Integer.parseInt(maxParallelSegmentPreprocessesBeforeServingQueriesStr);
-    } catch (Exception e) {
-      LOGGER.warn("Invalid maxPreprocessConcurrencyBeforeServingQueries set: {}, not making change, fix config and "
-              + "try again", maxParallelSegmentPreprocessesBeforeServingQueriesStr);
-      return;
-    }
-
-    if (maxPreprocessConcurrencyBeforeServingQueries <= 0) {
-      LOGGER.warn("maxPreprocessConcurrencyBeforeServingQueries: {} must be > 0, not making change, fix config "
-          + "and try again", maxPreprocessConcurrencyBeforeServingQueries);
-      return;
-    }
-
-    if (maxPreprocessConcurrencyBeforeServingQueries == _maxPreprocessConcurrencyBeforeServingQueries) {
-      LOGGER.info("No ZK update for maxPreprocessConcurrencyBeforeServingQueries {}, total permits: {}",
-          _maxPreprocessConcurrencyBeforeServingQueries, totalPermits());
-      return;
-    }
-
-    LOGGER.info("Updated maxPreprocessConcurrencyBeforeServingQueries from: {} to: {}",
-        _maxPreprocessConcurrencyBeforeServingQueries, maxPreprocessConcurrencyBeforeServingQueries);
-    _maxPreprocessConcurrencyBeforeServingQueries = maxPreprocessConcurrencyBeforeServingQueries;
-    if (!_isServingQueries) {
-      LOGGER.info("maxPreprocessConcurrencyBeforeServingQueries was updated before serving queries was enabled, "
-          + "updating the permits");
-      _semaphore.setPermits(_maxPreprocessConcurrencyBeforeServingQueries);
-      LOGGER.info("Updated total permits: {}", totalPermits());
-    }
-  }
-
-  /**
-   * Block trying to acquire the semaphore to perform the segment index rebuild steps unless interrupted.
-   * <p>
-   * {@link #release()} should be called after the segment preprocess completes. It is the responsibility of the caller
-   * to ensure that {@link #release()} is called exactly once for each call to this method.
-   *
-   * @throws InterruptedException if the current thread is interrupted
-   */
-  public void acquire()
-      throws InterruptedException {
-    _semaphore.acquire();
-  }
+  SegmentAllIndexPreprocessThrottler _segmentAllIndexPreprocessThrottler;
+  SegmentStarTreePreprocessThrottler _segmentStarTreePreprocessThrottler;
 
   /**
-   * Should be called after the segment index build completes. It is the responsibility of the caller to ensure that
-   * this method is called exactly once for each call to {@link #acquire()}.
+   * Constructor for SegmentPreprocessThrottler
+   * @param segmentAllIndexPreprocessThrottler segment preprocess throttler to use for all indexes
+   * @param segmentStarTreePreprocessThrottler segment preprocess throttler to use for StarTree index
    */
-  public void release() {
-    _semaphore.release();
+  public SegmentPreprocessThrottler(SegmentAllIndexPreprocessThrottler segmentAllIndexPreprocessThrottler,
+      SegmentStarTreePreprocessThrottler segmentStarTreePreprocessThrottler) {
+    LOGGER.info("Initializing SegmentPreprocessThrottler");
```

Comment:
This info is not very useful. I assume we are logging details within the individual throttlers?

Reply:
removed